### ISSUE TITLE
Code.format_string!/2 option `:locals_without_parens` not removing brackets?

### ISSUE THREAD

**Issue by neenjaw** [elixir-lang/elixir#9655]
### Precheck

- [x] Do not use the issue tracker for help or support (try Elixir Forum, Stack Overflow, IRC, etc.)
- [x] For proposing a new feature, please start a discussion on the Elixir Core mailing list: https://groups.google.com/group/elixir-lang-core
- [x] For bugs, do a quick search and make sure the bug has not yet been reported
- [x] Please disclose security vulnerabilities privately at elixir-security@googlegroups.com
- [x] Finally, be nice and have fun!

I was looking for some help on elixirforum for advice how to return a formatted code string from `Macro.to_string/1` because I was encountering some difficulty using `Code.format_string!/2` and the `:locals_without_parens` option.

[Following some discussion with @OvermindDL1 there](https://elixirforum.com/t/how-do-i-get-a-well-formatted-string-from-an-ast/19711/9?u=neenjaw), I think there may be a bug with the function.

### Environment

* Elixir & Erlang/OTP versions (elixir --version): 1.10.0-dev (OTP 22), 1.9.2 (OTP 22), 1.9.1 (OTP 22)
* Operating system: Ubuntu 18.04

### Current behavior

```shell
iex(1)> ast = quote do                                     
...(1)>   def add(a, b), do: a + b
...(1)> end
{:def, [context: Elixir, import: Kernel],
 [
   {:add, [context: Elixir], [{:a, [], Elixir}, {:b, [], Elixir}]},
   [
     do: {:+, [context: Elixir, import: Kernel],
      [{:a, [], Elixir}, {:b, [], Elixir}]}
   ]
 ]}
iex(2)> ast |> Macro.to_string
"def(add(a, b)) do\n  a + b\nend"
iex(3)> ast |> Macro.to_string |> Code.format_string!(locals_without_parens: [def: 1, def: 2, def: 3, def: :*])
["def", "(", "", "add", "(", "", "a", ",", " ", "b", "", ")", "", ")", " do",
 "\n  ", "a", " +", " ", "b", "\n", "end"]
iex(4)> ast |> Macro.to_string |> Code.format_string!(locals_without_parens: [def: 1, def: 2, def: 3, def: :*]) |> IO.puts()
def(add(a, b)) do
  a + b
end
```

### Expected behavior

From what I can gather from the documentation, I would expect the final `IO.puts` call to output:

```
def add(a, b) do
  a + b
end
```

rather than with the brackets surrounding the function declaration

**Comment by josevalim** [comment]
Local without parens do not remove parens. They only keep a call without parens if they were not there in the first place. The feature that you want has to be implemented in #9291. :)

**Comment by neenjaw** [comment]
Thanks for clarifying, @josevalim! At least I can rest easy knowing I'm not missing something obvious on my end!
